lib/util/git/index.ts

```diff
@@ -70,12 +71,31 @@ function checkForPlatformFailure(err: Error): void {
     'malformed object name',
     'TF401027:', // You need the Git 'GenericContribute' permission to perform this action
   ];
-  for (const errorStr of platformFailureStrings) {
+  for (const errorStr of externalHostFailureStrings) {
     if (err.message.includes(errorStr)) {
       logger.debug({ err }, 'Converting git error to ExternalHostError');
       throw new ExternalHostError(err, 'git');
     }
   }
+
+  const gitLabPushRuleFailureStrings = [
```

Comment:
I think this should be more generic, so we can reuse it for other platforms.

Reply:
Will `pushRuleFailureStrings` be ok?

Reply:
I think `configErrorStrings` is a better option. We can move some of the `externalHostFailureStrings` here.

* `TF401027:` 
* `remote: Invalid username or password`